rust-runtime/aws-smithy-runtime-api/src/client/orchestrator.rs

```diff
@@ -288,12 +265,12 @@ impl ConfigBagAccessors for ConfigBag {
         self.put::<Box<dyn RetryStrategy>>(Box::new(retry_strategy));
     }
 
-    fn request_time(&self) -> Option<RequestTime> {
-        self.get::<RequestTime>().cloned()
+    fn request_time(&self) -> Option<SharedTimeSource> {
+        self.get::<SharedTimeSource>().cloned()
     }
 
-    fn set_request_time(&mut self, request_time: RequestTime) {
-        self.put::<RequestTime>(request_time);
+    fn set_request_time(&mut self, request_time: impl TimeSource + 'static) {
+        self.put::<SharedTimeSource>(SharedTimeSource::new(request_time));
```

Comment:
Does this end up being a `Box<Arc<...>>` under the hood due to the `ConfigBag` implementation?

Reply:
yeah (but so does every thing we put in the ConfigBag, unfortunately)

there is a little trickery we could do to avoid that double-allocation with the new load/store traits because they can offer an enum that exposes the internals of the object so the config bag can use the smart pointer directly. not sure if it's worth it